matrix-sdk-android/src/test/java/org/matrix/android/sdk/internal/session/room/location/DefaultRedactLiveLocationShareTaskTest.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.android.sdk.internal.session.room.location
+
+import io.mockk.unmockkAll
+import io.realm.RealmList
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Test
+import org.matrix.android.sdk.internal.database.model.livelocation.LiveLocationShareAggregatedSummaryEntity
+import org.matrix.android.sdk.internal.database.model.livelocation.LiveLocationShareAggregatedSummaryEntityFields
+import org.matrix.android.sdk.test.fakes.FakeEventSenderProcessor
+import org.matrix.android.sdk.test.fakes.FakeLocalEchoEventFactory
+import org.matrix.android.sdk.test.fakes.FakeRealm
+import org.matrix.android.sdk.test.fakes.FakeRealmConfiguration
+import org.matrix.android.sdk.test.fakes.givenEqualTo
+import org.matrix.android.sdk.test.fakes.givenFindFirst
+
+private const val A_ROOM_ID = "room-id"
+private const val AN_EVENT_ID = "event-id"
+private const val AN_EVENT_ID_1 = "event-id-1"
+private const val AN_EVENT_ID_2 = "event-id-2"
+private const val AN_EVENT_ID_3 = "event-id-3"
+private const val A_REASON = "reason"
+
+@ExperimentalCoroutinesApi
+class DefaultRedactLiveLocationShareTaskTest {
+
+    private val fakeRealmConfiguration = FakeRealmConfiguration()
+    private val fakeLocalEchoEventFactory = FakeLocalEchoEventFactory()
+    private val fakeEventSenderProcessor = FakeEventSenderProcessor()
+    private val fakeRealm = FakeRealm()
+
+    private val defaultRedactLiveLocationShareTask = DefaultRedactLiveLocationShareTask(
+            realmConfiguration = fakeRealmConfiguration.instance,
+            localEchoEventFactory = fakeLocalEchoEventFactory.instance,
+            eventSenderProcessor = fakeEventSenderProcessor
+    )
+
+    @After
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun `given parameters when calling the task then it is correctly executed`() = runTest {
```

Comment:
this test name feels a little bit generic :thinking: if we match the implementation then we would get `when redacting then deletes event and related and creates redacted local echos` 


due to the amount of setup needed it makes the test a bit tricky to quickly read, what do you think about extracting some of the setup/verification?

```kotlin
val summary = createSummary(RELATED_EVENT_IDS) 
givenSummaryForEventId(AN_EVENT_ID, aggregatedSummaryEntity)
val redactedEchos = givenCreatesRedactedLocalEchos(AN_EVENT_ID + RELATED_EVENT_IDS) 

defaultRedactLiveLocationShareTask.execute(params)

verifyEventsRedacted(AN_EVENT_ID + RELATED_EVENT_IDS)
verifyLocalEchosCreated(redactedEchos)
verifyEventsRedacted(redactedEchos)
```

Reply:
Agree thanks for the suggestions. I did some refactoring on this test. See [131f97b](https://github.com/vector-im/element-android/pull/6486/commits/131f97b6310c57c171d525309a8aeeaabd38fda9)